Gems/PhysX/Code/Source/EditorMeshColliderComponent.h

```diff
@@ -0,0 +1,266 @@
+        void OnModelReady(const AZ::Data::Asset<AZ::RPI::ModelAsset>& modelAsset,
+            const AZ::Data::Instance<AZ::RPI::Model>& model) override;
+
+        // PhysX::ColliderShapeBus overrides ...
+        AZ::Aabb GetColliderShapeAabb() override;
+        bool IsTrigger() override;
+
+        // PhysX::EditorColliderComponentRequestBus overrides ...
+        void SetColliderOffset(const AZ::Vector3& offset) override;
+        AZ::Vector3 GetColliderOffset() const override;
+        void SetColliderRotation(const AZ::Quaternion& rotation) override;
+        AZ::Quaternion GetColliderRotation() const override;
+        AZ::Transform GetColliderWorldTransform() const override;
+        Physics::ShapeType GetShapeType() const override { return Physics::ShapeType::PhysicsAsset; }
+
+        // PhysX::EditorMeshColliderComponentRequestBus overrides ...
+        void SetAssetScale(const AZ::Vector3& scale) override;
+        AZ::Vector3 GetAssetScale() const override;
+
+        // PhysX::EditorMeshColliderValidationRequestBus overrides ...
+        void ValidateRigidBodyMeshGeometryType() override;
+
+        AZ::Transform GetColliderLocalTransform() const;
+
+        AZ::u32 OnConfigurationChanged();
+        void UpdateShapeConfigurationScale();
+
+        // AzPhysics::SimulatedBodyComponentRequestsBus::Handler overrides ...
+        void EnablePhysics() override;
+        void DisablePhysics() override;
+        bool IsPhysicsEnabled() const override;
+        AZ::Aabb GetAabb() const override;
+        AzPhysics::SimulatedBody* GetSimulatedBody() override;
+        AzPhysics::SimulatedBodyHandle GetSimulatedBodyHandle() const override;
+        AzPhysics::SceneQueryHit RayCast(const AzPhysics::RayCastRequest& request) override;
+
+        // Mesh collider
+        void UpdateMeshAsset();
+
+        void UpdateCollider();
+        void CreateStaticEditorCollider();
+
+        void BuildDebugDrawMesh() const;
+
+        AZ::ComponentDescriptor::StringWarningArray GetComponentWarnings() const { return m_componentWarnings; };
+
+        // Auto-assigning collision mesh utility functions
+        bool ShouldUpdateCollisionMeshFromRender() const;
+        void SetCollisionMeshFromRender();
+        AZ::Data::Asset<AZ::Data::AssetData> GetRenderMeshAsset() const;
+
+        AZ::Data::AssetId FindMatchingPhysicsAsset(
+            const AZ::Data::Asset<AZ::Data::AssetData>& renderMeshAsset, const AZStd::vector<AZ::Data::AssetId>& physicsAssets);
+
+        void ValidateAssetMaterials();
+
+        EditorProxyAssetShapeConfig m_shapeConfiguration;
+        Physics::ColliderConfiguration m_configuration;
+
+        using ComponentModeDelegate = AzToolsFramework::ComponentModeFramework::ComponentModeDelegate;
+        ComponentModeDelegate m_componentModeDelegate; //!< Responsible for detecting ComponentMode activation
+                                                       //!< and creating a concrete ComponentMode.
```

Comment:
It looks like this can be removed too as I don't think this Editor Component should have a component mode right?

Reply:
Editor Mesh Collider still needs to support component mode to help modify its `Offset`, `Rotation` parameters.

Reply:
I see good to know 👍